src/glue_launcher_lambda/glue_launcher.py

```diff
@@ -0,0 +1,517 @@
+    return boto3.client("batch")
+
+
+def get_athena_client():
+    return boto3.client("athena")
+
+
+def get_glue_client():
+    return boto3.service("glue")
+
+
+def check_running_batch_tasks(job_queue, batch_client):
+    """Check the AWS Batch job queue, return count of tasks in each status"""
+
+    operational_tasks = 0
+
+    for status in OPERATIONAL_JOB_STATUSES:
+        response = batch_client.list_jobs(
+            jobQueue=job_queue,
+            jobStatus=status,
+        )
+        operational_tasks += len(response["jobSummaryList"])
+    return operational_tasks
+
+
+def fetch_table_creation_sql_files(file_path, args=None):
+    with open(os.path.join(file_path, "create-parquet-table.sql"), "r") as f:
+        base_create_parquet_query = f.read()
+
+    with open(
+        os.path.join(file_path, "create-missing-import-table.sql"),
+        "r",
+    ) as f:
+        base_create_missing_import_query = f.read()
+
+    with open(
+        os.path.join(file_path, "create-missing-export-table.sql"),
+        "r",
+    ) as f:
+        base_create_missing_export_query = f.read()
+
+    with open(os.path.join(file_path, "create-count-table.sql"), "r") as f:
+        base_create_count_query = f.read()
+
+    tables = [
+        [
+            args.missing_imports_table_name,
+            base_create_missing_import_query,
+            args.manifest_s3_input_parquet_location_missing_import,
+        ],
+        [
+            args.missing_exports_table_name,
+            base_create_missing_export_query,
+            args.manifest_s3_input_parquet_location_missing_export,
+        ],
+        [
+            args.counts_table_name,
+            base_create_count_query,
+            args.manifest_s3_input_parquet_location_counts,
+        ],
+        [
+            args.mismatched_timestamps_table_name,
+            base_create_parquet_query,
+            args.manifest_s3_input_parquet_location_mismatched_timestamps,
+        ],
+    ]
+
+    return tables
+
+
+def fetch_table_drop_sql_file(file_path, args=None):
+    with open(os.path.join(file_path, "drop-table.sql"), "r") as f:
+        base_drop_query = f.read()
+        return base_drop_query
+
+
+def poll_athena_query_status(id):
+    """Polls athena for the status of a query.
+
+    Keyword arguments:
+    id -- the id of the query in athena
+    """
+    athena_client = get_client(service_name="athena")
+    time_taken = 1
+    while True:
+        query_execution_resp = athena_client.get_query_execution(QueryExecutionId=id)
+
+        state = query_execution_resp["QueryExecution"]["Status"]["State"]
+        if state in ("SUCCEEDED", "FAILED", "CANCELLED"):
+            console_printer.print_info(
+                f"Athena query execution finished in {str(time_taken)} seconds with status of '{state}'"
+            )
+            return state
+
+        time.sleep(1)
+        time_taken += 1
+
+
+def execute_athena_query(output_location, query, athena_client):
```

Comment:
I can't see a test for this method that ensures the client itself has the right calls